services/QuillLMS/app/controllers/auth/google_controller.rb

```diff
@@ -37,11 +40,11 @@ def authentication
     end
   end
 
-  private def verify_authorization_for_offline_access
-    user = User.where('google_id = ? OR email = ?', @profile.google_id&.to_s, @profile.email&.downcase).first
+  private def check_for_authorization
+    user = User.find_by('google_id = ? OR email = ?', @profile.google_id&.to_s, @profile.email&.downcase)
     return if user.nil?
 
-    redirect_to GoogleIntegration::AUTHORIZATION_AND_AUTHENTICATION_PATH unless user.google_authorized?
+    redirect_to new_session_path(google_offline_access_expired: true) unless user.google_authorized?
```

Comment:
Consider making this `return if user.google_authorized?` above

Reply:
Good idea